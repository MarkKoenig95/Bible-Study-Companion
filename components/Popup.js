import 'react-native-gesture-handler';
import React from 'react';
import {
  KeyboardAvoidingView,
  Platform,
  Text,
  StyleSheet,
  View,
} from 'react-native';
import IconButton from './IconButton';

import styles, {colors} from '../styles/styles';

export default function Popup(props) {
  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      style={[styles.background, {display: !props.displayPopup ? 'none' : ''}]}>
      <View
        style={{
          ...styles.popup,
          ...props.style,
        }}>
        <View style={style.title}>
          <Text style={style.text}>{props.title}</Text>
          <IconButton name="close" invertColor onPress={props.onClosePress} />
        </View>
        <View style={style.content}>{props.children}</View>
      </View>
    </KeyboardAvoidingView>
  );
}
const style = StyleSheet.create({
  closeButton: {
    backgroundColor: colors.lightBlue,
  },
  buttonText: {
    color: colors.darkGray,
    fontWeight: 'bold',
  },
  content: {
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'space-around',
    marginTop: 70,
    marginBottom: 10,
    width: '100%',
  },
  text: {
    ...styles.text,
    flex: 4,
    padding: 10,
    fontSize: 25,
    color: colors.darkGray,
  },
  title: {
    alignItems: 'center',
    backgroundColor: colors.smoke,
    flexDirection: 'row',
    height: 60,
    justifyContent: 'space-around',
    padding: 10,
    position: 'absolute',
    top: 0,
    width: '100%',
  },
});
