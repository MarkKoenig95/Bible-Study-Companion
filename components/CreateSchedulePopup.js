import React, {useState, useEffect} from 'react';
import {Keyboard} from 'react-native';

import IconButton from './IconButton';
import CustomInput from './CustomInput';
import VersePicker from './VersePicker';
import Popup from './Popup';

export default function CreateSchedulePopup(props) {
  useEffect(() => {
    Keyboard.addListener('keyboardWillShow', _keyboardWillShow);
    Keyboard.addListener('keyboardWillHide', _keyboardWillHide);

    // cleanup function
    return () => {
      Keyboard.removeListener('keyboardWillShow', _keyboardWillShow);
      Keyboard.removeListener('keyboardWillHide', _keyboardWillHide);
    };
  }, []);

  const _keyboardWillShow = () => {
    setMarginTop(10);
  };

  const _keyboardWillHide = () => {
    setMarginTop(100);
  };

  //State and defaults for shceudle info inputs
  const defaults = {
    scheduleName: '',
    scheduleDuration: '',
    chapter: '1',
    verse: '1',
    selectedItems: [],
  };

  const [scheduleName, setScheduleName] = useState(defaults.scheduleName);
  const [scheduleDuration, setScheduleDuration] = useState(
    defaults.scheduleDuration,
  );

  const [versePicker, setVersePicker] = useState({
    chapter: defaults.chapter,
    verse: defaults.verse,
    selectedItems: defaults.selectedItems,
  });

  const [marginTop, setMarginTop] = useState(null);

  function sanitizeNumber(prevValue, newValue, lowerLimit, upperLimit) {
    let result = '';

    newValue = newValue || '';
    prevValue = prevValue || '';

    let change = newValue.length - prevValue.length;

    if (change < 1) {
      result = newValue;
    } else {
      let number = parseInt(newValue, 10);
      if (!isNaN(number) && (number > lowerLimit && number < upperLimit)) {
        result = number.toString();
      } else {
        result = prevValue;
      }
    }

    return result;
  }

  function sanitizeLetter(prevValue, newValue) {
    let result = '';
    newValue = newValue || '';
    prevValue = prevValue || '';

    let change = newValue.length - prevValue.length;

    if (change < 1) {
      result = newValue;
    } else {
      let newChar = newValue.slice(newValue.length - 1);

      if (
        (newChar >= 'a' && newChar <= 'z') ||
        (newChar >= 'A' && newChar <= 'Z')
      ) {
        result = newValue;
      } else {
        result = prevValue;
      }
    }

    return result;
  }

  function onVersePickerChange(key, value) {
    if (key !== 'selectedItems') {
      value = sanitizeNumber(versePicker[key], value, 0, 201);
    }

    setVersePicker(prevVals => {
      return {...prevVals, [key]: value};
    });
  }

  function onScheduleNameChange(text) {
    let sanitizedState = sanitizeLetter(scheduleName, text);
    setScheduleName(sanitizedState);
  }

  function onScheduleDurationChange(text) {
    let sanitizedState = sanitizeNumber(scheduleDuration, text, 0, 21);
    setScheduleDuration(sanitizedState);
  }

  function onAddPress() {
    //TODO: Check input values to validate
    if (
      versePicker.selectedItems[0].id &&
      scheduleName &&
      scheduleDuration &&
      versePicker.chapter &&
      versePicker.verse
    ) {
      let bookId = versePicker.selectedItems[0].id;

      props.onAdd(
        scheduleName,
        scheduleDuration,
        bookId,
        versePicker.chapter,
        versePicker.verse,
      );
      // setScheduleName(defaults.scheduleName);
      // setScheduleDuration(defaults.scheduleDuration);
      // setVersePicker({
      //   chapter: defaults.chapter,
      //   verse: defaults.verse,
      //   selectedItems: versePicker.selectedItems,
      // });
    } else {
      props.onError(
        'Please fill in all of the required fields to make a schedule',
      );
    }
  }

  return (
    <Popup
      style={marginTop && {marginTop: marginTop}}
      displayPopup={props.displayPopup}
      title="Create Schedule"
      onClosePress={props.onClosePress}>
      <CustomInput
        title="Schedule Name"
        onChangeText={text => onScheduleNameChange(text)}
        value={scheduleName}
        placeholder="Schedule Name"
      />
      <CustomInput
        title="Schedule Duration"
        onChangeText={text => onScheduleDurationChange(text)}
        value={scheduleDuration}
        placeholder="In Years"
      />
      <VersePicker
        title="Starting Verse"
        onChange={onVersePickerChange}
        selectedItems={versePicker.selectedItems}
        chapterValue={versePicker.chapter}
        verseValue={versePicker.verse}
      />
      <IconButton name="add" onPress={onAddPress} />
    </Popup>
  );
}
