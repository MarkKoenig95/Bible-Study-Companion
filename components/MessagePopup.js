import React from 'react';
import {Text, StyleSheet} from 'react-native';

import Popup from './Popup';
import IconButton from './IconButton';

import styles from '../styles/styles';

export default function MessagePopup(props) {
  const hasConfirmButton = props.onConfirm && true;

  return (
    <Popup
      displayPopup={props.displayPopup}
      title={props.title}
      onClosePress={props.onClosePress}>
      <Text style={style.text}>{props.message}</Text>
      {hasConfirmButton && (
        <IconButton name="check" onPress={props.onConfirm} />
      )}
    </Popup>
  );
}

const style = StyleSheet.create({
  text: {...styles.text, fontSize: 30, padding: 20},
});
