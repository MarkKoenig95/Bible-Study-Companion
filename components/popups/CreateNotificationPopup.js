import React, {useState} from 'react';
import {View, StyleSheet} from 'react-native';

import styles, {colors} from '../../styles/styles';
import Popup from './Popup';
import CustomInput from '../inputs/CustomInput';
import {translate} from '../../logic/localization/localization';
import IconButton from '../buttons/IconButton';
import Text from '../text/Text';
import CheckBox from '../buttons/CheckBox';
import TimePickerButton from '../buttons/TimePickerButton';

function WeekdayCheckbox(props) {
  const {abrev, checked, id, onPress, testID} = props;
  return (
    <View style={style.weekdayCheckboxContainer}>
      <Text
        style={{
          ...style.weekdayAbrevText,
          color: !checked ? styles.lightText.color : colors.darkBlue,
        }}>
        {abrev}
      </Text>
      <CheckBox
        testID={testID}
        checked={checked}
        uncheckedColor={styles.lightText.color}
        checkedColor={colors.darkBlue}
        onPress={() => {
          onPress(id, !checked);
        }}
      />
    </View>
  );
}

export default function CreateNotificationPopup(props) {
  const {displayPopup, prefix, onAddPress, onClosePress, testID, title} = props;
  const [time, setTime] = useState(new Date(2020, 0, 1, 8, 0, 0));
  const [notificationName, setNotificationName] = useState('');
  const [dayValues, setDayValues] = useState({
    0: true,
    1: true,
    2: true,
    3: true,
    4: true,
    5: true,
    6: true,
  });

  const days = [];
  const weekdays = [];

  for (let i = 0; i < 7; i++) {
    const abrev = translate('weekdays.' + i + '.abrev');
    weekdays.push({id: i, abrev: abrev});
    days.push(dayValues[i]);
  }

  return (
    <Popup
      testID={testID}
      displayPopup={displayPopup}
      title={title}
      onClosePress={onClosePress}>
      <CustomInput
        testID={testID + '.nameInput'}
        title={translate(prefix + 'notificationName')}
        onChangeText={setNotificationName}
        value={notificationName}
        placeholder={translate(prefix + 'notificationName')}
      />
      <View style={styles.wrapperContent}>
        {weekdays.map(day => {
          return (
            <WeekdayCheckbox
              testID={testID + '.weekdayCheckbox.' + day.abrev}
              key={Math.random() * 1000000}
              abrev={day.abrev}
              id={day.id}
              checked={dayValues[day.id]}
              onPress={(id, value) => {
                setDayValues({...dayValues, [id]: value});
              }}
            />
          );
        })}
      </View>

      <TimePickerButton
        testID={testID + '.timePicker'}
        textPrefix={translate('prompts.setNotification')}
        time={time}
        onChange={setTime}
      />

      <IconButton
        testID={testID + '.addButton'}
        name="add"
        onPress={() => {
          onAddPress(notificationName, days, time);
        }}
      />
    </Popup>
  );
}

export function useCreateNotificationPopup() {
  const openNotificationPopup = () => {
    setNotificationPopup({
      ...notificationPopup,
      isDisplayed: true,
    });
  };

  const closeNotificationPopup = () => {
    setNotificationPopup({
      ...notificationPopup,
      isDisplayed: false,
    });
  };

  const [notificationPopup, setNotificationPopup] = useState({
    isDisplayed: false,
    title: translate('notificationsPage.notificationPopupTitle'),
    open: openNotificationPopup,
    close: closeNotificationPopup,
  });

  return {notificationPopup};
}

const style = StyleSheet.create({
  weekdayCheckboxContainer: {
    alignItems: 'center',
    flex: 1,
    margin: 0,
    padding: 0,
  },
  weekdayAbrevText: {
    fontSize: 15,
    margin: 0,
    padding: 0,
  },
});
