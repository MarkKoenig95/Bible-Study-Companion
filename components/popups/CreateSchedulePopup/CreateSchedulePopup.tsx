import React from 'react';

import IconButton from '../../buttons/IconButton';
import CustomInput from '../../inputs/CustomInput';
import Popup from '../Popup';
import {
  AdvancedSettingsSection,
  CustomInputs,
  ScheduleInputs,
} from './components';

import {translate} from '../../../logic/localization/localization';
import {useCreateSchedulePopup} from './logic';
import {CreateSchedulePopupProps} from './types';

const prefix = 'createSchedulePopup.';

export default function CreateSchedulePopup(props: CreateSchedulePopupProps) {
  const {displayPopup, onClosePress, testID} = props;
  const {
    doesTrack,
    isCustom,
    maxPortion,
    onAddPress,
    onScheduleDurationChange,
    onVersePickerChange,
    portionsPerDay,
    readingPortionDesc,
    readingPortionSelectedItems,
    scheduleDuration,
    scheduleName,
    setDoesTrack,
    setMaxPortion,
    setPortionsPerDay,
    setReadingPortionDesc,
    setReadingPortionSelectedItems,
    setScheduleName,
    setStartDate,
    setStartingPortion,
    startDate,
    startingPortion,
    versePicker,
  } = useCreateSchedulePopup(props);

  return (
    <Popup
      testID={testID}
      displayPopup={displayPopup}
      title={translate(prefix + 'createSchedule')}
      onClosePress={onClosePress}>
      <CustomInput
        testID={testID + '.scheduleNameInput'}
        title={translate(prefix + 'scheduleName')}
        onChangeText={setScheduleName}
        value={scheduleName}
        placeholder={translate(prefix + 'scheduleName')}
      />
      {isCustom ? (
        <CustomInputs
          testID={testID}
          maxPortion={maxPortion}
          portionsPerDay={portionsPerDay}
          readingPortionDesc={readingPortionDesc}
          readingPortionSelectedItems={readingPortionSelectedItems}
          setMaxPortion={setMaxPortion}
          setPortionsPerDay={setPortionsPerDay}
          setReadingPortionDesc={setReadingPortionDesc}
          setReadingPortionSelectedItems={setReadingPortionSelectedItems}
          setStartingPortion={setStartingPortion}
          startingPortion={startingPortion}
        />
      ) : (
        <ScheduleInputs
          testID={testID}
          onScheduleDurationChange={onScheduleDurationChange}
          scheduleDuration={scheduleDuration}
          onVersePickerChange={onVersePickerChange}
          selectedItems={versePicker.selectedItems}
          chapterValue={versePicker.chapter}
          verseValue={versePicker.verse}
        />
      )}

      <AdvancedSettingsSection
        testID={testID}
        doesTrack={doesTrack}
        setDoesTrack={setDoesTrack}
        setStartDate={setStartDate}
        startDate={startDate}
      />

      <IconButton
        testID={testID + '.addButton'}
        name="add"
        onPress={onAddPress}
      />
    </Popup>
  );
}
