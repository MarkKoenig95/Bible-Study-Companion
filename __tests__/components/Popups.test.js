import React from 'react';
import renderer from 'react-test-renderer';
import CreateNotificationPopup from '../../components/popups/CreateNotificationPopup';
import CreateReminderPopup from '../../components/popups/CreateReminderPopup';
import CreateSchedulePopup from '../../components/popups/CreateSchedulePopup';
import LoadingPopup from '../../components/popups/LoadingPopup';
import MessagePopup from '../../components/popups/MessagePopup';
import Popup from '../../components/popups/Popup';
import ReadingInfoPopup from '../../components/popups/ReadingInfoPopup';
import ReadingRemindersPopup from '../../components/popups/ReadingRemindersPopup';
import ScheduleTypeSelectionPopup from '../../components/popups/ScheduleTypeSelectionPopup';
import SelectedDayButtonsPopup from '../../components/popups/SelectedDayButtonsPopup';

import {StateProvider} from '../../data/Store/store';

jest.mock('../../data/Store/store');

beforeAll(() => {
  // Tell Jest to use a different timer implementation. You can also
  // configure this in your jest.config.js file. For more info see
  // https://jestjs.io/docs/en/configuration#timers-string).
  jest.useFakeTimers('modern');

  jest.setSystemTime(new Date('04 Feb 2021 00:12:00 GMT').getTime());
});

afterAll(() => {
  jest.useRealTimers();
});

describe('testing if popups render', () => {
  it('renders CreateNotificationPopup correctly', () => {
    renderer.create(<CreateNotificationPopup />);
  });
  it('renders CreateReminderPopup correctly', () => {
    renderer.create(<CreateReminderPopup />);
  });
  it('renders CreateSchedulePopup correctly', () => {
    renderer.create(
      <StateProvider>
        <CreateSchedulePopup />
      </StateProvider>,
    );
  });
  it('renders LoadingPopup correctly', () => {
    renderer.create(<LoadingPopup />);
  });
  it('renders MessagePopup correctly', () => {
    renderer.create(<MessagePopup />);
  });
  it('renders Popup and matches snapshot', () => {
    const tree = renderer.create(<Popup />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders ReadingInfoPopup correctly', () => {
    renderer.create(
      <StateProvider>
        <ReadingInfoPopup />
      </StateProvider>,
    );
  });
  it('renders ReadingRemindersPopup and matches snapshot', () => {
    const tree = renderer.create(<ReadingRemindersPopup />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders ScheduleTypeSelectionPopup correctly', () => {
    renderer.create(<ScheduleTypeSelectionPopup />);
  });
  it('renders SelectedDayButtonsPopup correctly', () => {
    renderer.create(<SelectedDayButtonsPopup buttons={[]} />);
  });
});
