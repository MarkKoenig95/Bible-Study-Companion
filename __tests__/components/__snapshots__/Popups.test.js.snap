// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing if popups render renders ReadingRemindersPopup and matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#93b5b360",
        "elevation": 3,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
        "zIndex": 2,
      },
      Object {
        "display": "none",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#93b5b3",
          "borderRadius": 10,
          "maxHeight": "95%",
          "minHeight": 300,
          "overflow": "hidden",
          "width": "85%",
          "zIndex": 2,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f2f6f5",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-around",
          "padding": 5,
          "paddingLeft": 20,
          "paddingRight": 20,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {},
            Object {
              "color": "#f2f6f5",
            },
            Object {
              "color": "#63707e",
              "flex": 4,
              "fontSize": 20,
              "padding": 10,
            },
          ]
        }
      >
        Reading Reminders
      </Text>
      <View>
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#93b5b3",
              "borderRadius": 25,
              "elevation": 0,
              "height": 50,
              "margin": 10,
              "opacity": 1,
              "padding": 10,
              "shadowColor": "#0f7a72",
              "shadowOffset": Object {
                "height": 6,
                "width": 6,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 3,
              "width": 50,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#f2f6f5",
                  "fontSize": 30,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "space-around",
          "maxWidth": "100%",
          "padding": 10,
        }
      }
      keyboardShouldPersistTaps="handled"
      style={
        Object {
          "alignContent": "center",
          "flexDirection": "column",
          "marginBottom": 10,
          "marginTop": 60,
          "width": "100%",
        }
      }
    >
      <View>
        <View>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "padding": 10,
                  "paddingTop": 20,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Create the right surroundings
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Find a location that is quiet and free from distractions
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            (Perhaps put your phone into silent mode)
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "padding": 10,
                  "paddingTop": 20,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Pray before you read
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Ask Jehovah to put you in the proper frame of mind and to give you his Holy Spirit to help you understand and apply what you read.
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "padding": 10,
                  "paddingTop": 20,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Meditate on what you read
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Don’t rush, read to understand.
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "padding": 10,
                  "paddingTop": 20,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            You could ask these questions:
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            What does this tell me about Jehovah God?
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            How does this section of the Scriptures contribute to the Bible’s message?
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            How can I apply this in my life?
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            How can I use these verses to help others?
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "padding": 10,
                  "paddingTop": 20,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Imagine yourself in the scene:
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            What would you see, hear, and smell?
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            What may be the feelings of those involved?
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "padding": 10,
                  "paddingTop": 20,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Do research in order to:
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Understand the reading portion’s context.
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Understand difficult concepts clearly.
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 25,
                  "fontWeight": "bold",
                  "padding": 10,
                  "paddingTop": 20,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Apply what you read
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            If we see the practical value of an activity, we benefit more from it.
          </Text>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            We need to take action to apply what we learned in our life and understand how we can use what we learned to help others.
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "padding": 10,
                  "paddingTop": 20,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Reminders derived from:
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "color": "#f2f6f5",
                  },
                  Array [
                    Object {
                      "color": "#0f7a72",
                      "fontSize": 20,
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                    },
                    undefined,
                  ],
                ]
              }
            >
              nwt 36; 
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "color": "#f2f6f5",
                  },
                  Array [
                    Object {
                      "color": "#0f7a72",
                      "fontSize": 20,
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                    },
                    undefined,
                  ],
                ]
              }
            >
              mwb18.03 8; 
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "color": "#f2f6f5",
                  },
                  Array [
                    Object {
                      "color": "#0f7a72",
                      "fontSize": 20,
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                    },
                    undefined,
                  ],
                ]
              }
            >
              w12 1/15 15; 
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "color": "#f2f6f5",
                  },
                  Array [
                    Object {
                      "color": "#0f7a72",
                      "fontSize": 20,
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                    },
                    undefined,
                  ],
                ]
              }
            >
              wp17.1 4; 
            </Text>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {
                    "color": "#f2f6f5",
                  },
                  Array [
                    Object {
                      "color": "#0f7a72",
                      "fontSize": 20,
                      "paddingBottom": 10,
                      "paddingLeft": 10,
                    },
                    undefined,
                  ],
                ]
              }
            >
              w16.05 24-26; 
            </Text>
          </View>
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "flex-start",
                  "color": "#f2f6f5",
                  "fontSize": 18,
                  "padding": 10,
                },
                Object {
                  "color": "#f2f6f5",
                },
                undefined,
              ]
            }
          >
            Please go to those links to get a deeper understanding of the thoughts expressed
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
